### PR TITLE
Move reusable pieces to chat-core

### PR DESCRIPTION
**Pull request by cbush**
- Creates a package `chat-core` and moves many things from `chat-server` into it so they can be reused from other places, namely `ingest`.

### CODE REVIEW

File: chat-server/src/aiConstants.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+export const ASSISTANT_PROMPT: OpenAiChatMessage = {
+  role: "assistant",
+  content: "Hello, I'm a MongoDB chatbot. How can I help you today?",
+};
+export const OPENAI_LLM_CONFIG_OPTIONS: GetChatCompletionsOptions = {
+  temperature: 0.3,
+  maxTokens: 500,
+};
+
+interface GenerateUserPromptParams {
```
Comment (mongodben) [inline chat-server/src/aiConstants.ts:30]:
```suggestion
export interface GenerateUserPromptParams {
```

File: chat-server/package.json [hunk 2]
```diff
@@ -22,16 +22,16 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
-    "@azure/openai": "^1.0.0-beta.2",
+    "chat-core": "file:../chat-core",
```
Comment (mongodben) [inline chat-server/package.json:25]: this is easier than i expected! thought would need some npm link or smthn like that

File: chat-server/seed-content/index.ts [hunk 3]
```diff
@@ -4,9 +4,7 @@ import { Content } from "../src/services/content";
 import { RecursiveCharacterTextSplitter } from "langchain/text_splitter";
 import { Document } from "langchain/dist/document";
 import GPT3Tokenizer from "gpt3-tokenizer";
-import { embeddings } from "../src/services/embeddings";
-import { ObjectId } from "mongodb";
-import { mongodb } from "../src/integrations/mongodb";
+import { ObjectId, embeddings, mongodb } from "chat-core";
```
Comment (mongodben) [inline chat-server/seed-content/index.ts:7]: as we discussed before, we should refactor these to not be exported singletons..but subject of a future PR there

File: chat-server/src/aiConstants.ts
Comment (mongodben) [inline chat-server/src/aiConstants.ts:1]: nit: can you rename `openAiConstants.ts`? these are specific to our OpenAI API implementation, and would presumably be different if we are using another AI service, like Anthropic (or even something in-house)

at some point, we could create abstractions here that'd work for all AI services (perhaps using or inspired by [LangChain](https://js.langchain.com/docs/)), but that's not what we're doing now.

Comment (cbush) [reply chat-server/src/aiConstants.ts:1]: I don't think openAiConstants.ts will survive long anyway - they aren't supposed to be constants, we should be able to swap them in and out to compare and test etc.

File: chat-server/seed-content/index.ts (hunk 3 above)
Comment (mongodben) [inline chat-server/seed-content/index.ts:7]: as we discussed before, we should refactor these to not be exported singletons..but subject of a future PR there.

File: chat-server/src/aiConstants.ts
Comment (mongodben) [reply chat-server/src/aiConstants.ts:1]: how do you envision them? as env variables?

**Review by mongodben — APPROVED** [review]

File: chat-server/src/aiConstants.ts
Comment (cbush) [reply chat-server/src/aiConstants.ts:1]: Maybe. I would put all the prompts together somehow and choose which one to pass in based on some configuration.

**Pull request merged**